### PR TITLE
enhancement(dev): k8s e2e tests script minor improvements

### LINKED ISSUE #7545: Kubernetes end to end tests should run on Macs.
Currently the tests assume that you are using Linux. This allows them to build Vector locally and deploy a container to test to Docker Hub. 

On a Mac you have to deploy the Vector container to Docker Hub yourself and then specify `SKIP_CONTAINER_IMAGE_PUBLISHING=true CONTAINER_IMAGE=<image>` when running the tests. We should build it into the tests to use Docker to build a Vector image for Macs. There is also a possibility we could use cross compiling, but I couldn't get it to work - there were issues with some of the third party libraries we use.

### PR DESCRIPTION
**Pull request by prognant**
closes #7545

Account for minikube/kind non-default cluster name and
switch from `minikube cache` to `minikube image` as
the former is scheduled for deprecation.

### CODE REVIEW

File: scripts/test-e2e-kubernetes.sh [hunk 1]
```diff
@@ -121,8 +127,8 @@ fi
 if [[ -z "${SKIP_CONTAINER_IMAGE_PUBLISHING:-}" ]]; then
   # Make the container image accessible to the k8s cluster.
   if is_minikube_cache_enabled; then
-    minikube cache add "$CONTAINER_IMAGE"
-    trap 'minikube cache delete "$CONTAINER_IMAGE"' EXIT
+    minikube image load "$CONTAINER_IMAGE"
+    trap 'minikube image rm "$CONTAINER_IMAGE"' EXIT
```
Comment (spencergilbert) [inline scripts/test-e2e-kubernetes.sh:131]: `minikube image` is the new syntax right? We'll need to make sure the existing version used in CI supports it, or (preferably in my opinion) upgrade the minikube used in CI

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: scripts/test-e2e-kubernetes.sh (hunk 1 above)
Comment (spencergilbert) [reply scripts/test-e2e-kubernetes.sh:131]: I just confirmed our minikube version in CI _doesn't_ have the `image` subcommand

Comment (prognant) [reply scripts/test-e2e-kubernetes.sh:131]: Indeed, I push a naive upgrade attempt, let see if CI is happy with it

Comment (prognant) [reply scripts/test-e2e-kubernetes.sh:131]: The CI is all green, I don't thing there is any reason not to upgrade minikube.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**